core/net.go

```diff
@@ -188,8 +190,10 @@ func (n *OpenBazaarNode) SendOrderConfirmation(peerId string, contract *pb.Ricar
 	}
 	err = n.Service.SendMessage(ctx, p, &m)
 	if err != nil {
-		if err := n.SendOfflineMessage(p, &m); err != nil {
-			return err
+		if k, err := libp2p.UnmarshalPublicKey(contract.GetBuyerOrder().GetBuyerID().GetPubkeys().Guid); err != nil {
```

Comment:
The error handling seems wrong here. If there is an error with `UnmarshalPublicKey`, return err. If not, move on to `SendOfflineMessage`. This should also avoid one level of code nesting. Similar issues below.